### DISCUSSION TITLE
How can predbat better understand when the car is charging

### DISCUSSION THREAD

**Discussion by marcoaddario** [springfall2008/batpred#463] [Q&A]
Just a question for the moment, as it is maybe already possible to manage my actual setup (or maybe is already managed)
I have recently switch from the car-controlled Intelligent Octopus to the charger-controlled one. My charger is a myenergi zappi.
When IO controlled the car, they had the exact SOC of the car. With the control moved to the charger, they don't know when the car is fully charged and the charge slot may be bigger than the effective charge time.
For example tonight I have a planned charge slot from 23:00 to 05:30 while, according to the car, it should finish around 3:30.
Now, my question is: will predbat get the real charge load or it will trust the IO charge slot? and how this can affect the prediction?

**Comment by RobinCu** [comment]
With the IOG via zappi, if switch.predbat_octopus_intelligent_charging is enabled, then Predbat will trust the IO charge slot (from the BottleCapDave integration).  
So if you choose "Charge to add" of 100%, ready by 09:00, Octopus might determine a 20:00 to 09:00 off-peak slot, and Dave's integration will pick up the cheap rate dispatches for that time.  Predbat will assume the car will charge for the entire duration and plan accordingly.  
I've found that by having switch.predbat_octopus_intelligent_charging off, Predbat works using my car's SoC sensor (as defined in apps.yml) and plans much better.  switch.predbat_octopus_intelligent_charging doesn't impact on switch.octopus_energy_[account number]_intelligent_smart_charge, which obviously needs to be on to get the extra charge slots.
What Predbat can never know is when Octopus' Kraken system will decide to start/stop charging, but with the zappi integration, it seems to map the slots and start almost immediately unless there is a really high demand on the grid.
Edit: to increase the accuracy, I also use a input helper for charge limit so Predbat knows the true charge limit set in the car.  It can then determine charge limit minus Soc x charger rate and calculate charge time.
